spec/unit/classes/server_spec.rb

```diff
@@ -157,4 +157,78 @@ class { 'postgresql::globals':
       is_expected.to contain_class('postgresql::repo').with_version('99.5')
     end
   end
+
+  describe 'create config entries' do
+    let(:params) {{ 'config_entries' => {'config_entry' => {'ensure' => 'present', 'name' => 'port_spec', 'value' => '5432'}}}}
+    it 'stops postgresql and changes the port' do
+      is_expected.to contain_file('systemd-override')
+      is_expected.to contain_exec('restart-systemd')
+    end
+  end
+
+  describe 'create database entries' do
+    let(:params) {{ 'databases' => {'test' => {'comment' => 'test comment'}} }}
+
+    it { is_expected.to contain_postgresql_psql("COMMENT ON DATABASE \"test\" IS 'test comment'").with_connect_settings( {} ) }
+  end
+
+  describe 'create database_grant entries' do
+    let(:params) {{ 'database_grants' => {'test' => {'privilege' => 'ALL', 'db' => 'test', 'role' => 'test'}} }}
+
+    it { is_expected.to contain_postgresql__server__database_grant('test') }
+    it { is_expected.to contain_postgresql__server__grant('database:test') }
+  end
+
+  describe 'create db entries' do
+    let(:params) {{ 'dbs' => {'test' => {'user' => 'test', 'password' => 'test', 'owner' => 'tester'}} }}
+
+    it { is_expected.to contain_postgresql__server__db('test') }
+    it { is_expected.to contain_postgresql__server__database('test').with_owner('tester') }
+    it { is_expected.to contain_postgresql__server__role('test').that_comes_before('Postgresql::Server::Database[test]') }
+    it { is_expected.to contain_postgresql__server__database_grant('GRANT test - ALL - test') }
+  end
+
+  describe 'create pg_hba_rule entries' do
+    let(:params) {{ 'pg_hba_rules' => {'test' => {'type' => 'host', 'database' => 'all', 'user' => 'all', 'address' => '1.1.1.1/24', 'auth_method' => 'md5', 'target' => tmpfilename('pg_hba_rule')}} }}
+
+    it { is_expected.to contain_concat__fragment('pg_hba_rule_test') }
+  end
+
+  describe 'create pg_ident_rule entries' do
+    let(:params) {{ 'pg_ident_rules' => {'test' => {'map_name' => 'thatsmymap', 'system_username' => 'systemuser', 'database_username' => 'dbuser', }} }}
+
+    it { is_expected.to contain_concat__fragment('pg_ident_rule_test') }
+  end
+
+  describe 'create recovery entries' do
+    let(:params) {{ 'recovery' => {'test' => {'restore_command' => 'restore_command', 'recovery_target_timeline' => 'recovery_target_timeline' }} }}
```

Comment:
I think you also need to add `'manage_recovery_conf' => true` to the params hash for the recovery feature to activate.